source/velocity_boundary_conditions/gplates.cc

```diff
@@ -220,207 +198,102 @@ namespace aspect
-          residual_normal_velocity = (surf_vel * position) / (surf_vel.norm() * position.norm());
-
-        AssertThrow( residual_normal_velocity < 1e-11,
-                     ExcMessage("Error in velocity boundary module interpolation. "
-                                "Radial component of velocity should be zero, but is not."));
-
-        // After checking that the residual normal velocity is small
-        // we might as well remove it to increase compatibility of the
-        // pressure right hand side.
-        const Tensor<1,3> normalized_position = position / position.norm();
-        surf_vel -=  (surf_vel * normalized_position) * normalized_position;
-
-        // We have interpolated over the whole dataset of the provided velocities, which is ok.
-        // Velocity will be constant for all evaluation points in this case.
-        return surf_vel / n_interpolation_weight;
-      }
-
-      double
-      GPlatesLookup::add_interpolation_point(Tensor<1,3>       &surf_vel,
-                                             const Tensor<1,3> &position,
-                                             const int          spatial_index[2],
-                                             const double       time_weight,
-                                             const bool         check_termination) const
-      {
-        // If the point is extended over the poles, do not use it. It will be found
-        // by the check in phi direction.
-        if ((spatial_index[0] < 0) || (spatial_index[0] >= static_cast<int>(velocity_values->n_rows())))
-          return 0.0;
-
-        int idx[2] = {spatial_index[0],spatial_index[1]};
-        reformat_indices(idx);
-
-        const Tensor<1,3> spherical_point = get_grid_point_position(idx[0],idx[1],false);
-        const Tensor<1,3> cartesian_point = cartesian_surface_coordinates(spherical_point);
-
-        const double arc_dis = arc_distance(position,position.norm() * cartesian_point);
+        // Main work, interpolate velocity at this point
+        Tensor<1,2> interpolated_velocity;
 
-        // termination criterion, our interpolation points are outside of the defined circle
-        if (check_termination && (arc_dis > interpolation_width))
+        for (unsigned int i = 0; i < 2; i++)
           {
-            return -1;
+            interpolated_velocity[i] = velocities[i]->value(internal_position_in_spher_rad);
           }
 
-        // sin(theta) accounts for the different area covered by grid points at varying latitudes
-        // the minimal value is chosen to weight points at the poles according to their number and area
-        const double point_weight = std::max(std::sin(spherical_point[0]),
-                                             std::sin(delta_theta/2)/velocity_values->n_cols());
-
-        const Tensor<1,3> normalized_position = position / position.norm();
+        //transform interpolated_velocity in cartesian coordinates
+        Tensor<1,3> interpolated_velocity_in_cart;
```

Comment:
you can combine the following two lines and make interpolated_velocity_in_cart `const`